File: examples/EnableAgentServiceDemo.java
package com.cisco.ccbu.cce.unifiedconfig.toolkit.examples;


import com.cisco.ccbu.cce.unifiedconfig.toolkit.RESTClient;
import com.cisco.ccbu.cce.unifiedconfig.toolkit.bean.Agent;


/**
 * Example code for creating agent with agent service enabled.
 *
 * Command line launch: mvn exec:java -Dexec.mainClass="com.cisco.ccbu.cce.unifiedconfig.toolkit.examples.EnableAgentServiceDemo" -Dexec.args="10.86.135.193 administrator@test.com abc123="
 */
public class EnableAgentServiceDemo {

    public static void main(String[] args) {

        if(args.length < 3){
            System.out.println("EnableAgentServiceDemo requires 3 parameters: hostname username (with @domain) password");
            System.exit(0);
        }

        // Create a new RESTClient object with the IP of AW HDS
        RESTClient restClient = new RESTClient(args[0], args[1], args[2]);
        DemoUtils.checkPCCEDeploymentType(restClient);

        try {
            enableAgentServiceDemo(restClient);
        } catch (Exception e){
            System.out.println("Exception occured: "+e.getMessage());
        }
    }

    private static void enableAgentServiceDemo(RESTClient restClient) throws ClassNotFoundException {
        // Create a new agent with 3 agent services enabled, "Transcript", "AgentAnswers", "VAVTranscript"
        String payload = "{\"agentServicesEnabled\": [{\"agentService\": [\"AgentAnswers\",\"Transcript\", \"VAVTranscript\"]}]," +
                "    \"person\": {" +
                "        \"ecePerson\": false," +
                "        \"firstName\": \""+DemoUtils.generateUniqueString("firstName", 1)+"\"," +
                "        \"lastName\": \""+DemoUtils.generateUniqueString("lastName", 1)+"\"," +
                "        \"loginEnabled\": true," +
                "        \"ssoEnabled\": false," +
                "        \"userName\": \""+DemoUtils.generateUniqueString("userName", 1)+"\"" +
                "    }" +
                "}";

        // -- Create and get the agent
        Agent newAgent = restClient.createAndGet_JSONPayload(payload, "agent", Class.forName("com.cisco.ccbu.cce.unifiedconfig.toolkit.bean.Agent"));
        System.out.println("Agent created: " + newAgent.getRefURL());
        System.out.println("Agent Services enabled for the agent: " + newAgent.getAgentServicesEnabled());

        // delete the agent
        restClient.delete(newAgent.getRefURL());
        System.out.println("Deleted Agent: " + newAgent.getRefURL());
    }
}

File: CiscoPCCE.Toolkit/Examples/EnableAgentServiceDemo.cs
using CiscoPCCE.Toolkit.Bean;
using CiscoPCCE.Toolkit.Sdk;

namespace CiscoPCCE.Toolkit.Examples
{
    /// <summary>
    /// Example code for creating agent with agent service enabled.
    /// </summary>
    public class EnableAgentServiceDemo
    {
        public static async Task Main(string[] args)
        {
            if (args.Length < 3)
            {
                Console.WriteLine("EnableAgentServiceDemo requires 3 parameters: hostname username (with @domain) password");
                Environment.Exit(0);
            }

            // Create a new RESTClient object with the IP of AW HDS
            var restClient = new CiscoRestClient(args[0], args[1], args[2]);
            await DemoUtils.CheckPCCEDeploymentTypeAsync(restClient);

            try
            {
                await EnableAgentServiceDemoAsync(restClient);
            }
            catch (Exception e)
            {
                Console.WriteLine($"Exception occurred: {e.Message}");
            }
        }

        private static async Task EnableAgentServiceDemoAsync(CiscoRestClient restClient)
        {
            // Create a new agent with 3 agent services enabled
            // Note: Based on the C# Bean structure, we use MappedServiceType
            var newAgent = new Agent
            {
                AgentServicesEnabled = new List<AgentServiceEnabled>
                {
                    new AgentServiceEnabled { MappedServiceType = "AgentAnswers" },
                    new AgentServiceEnabled { MappedServiceType = "Transcript" },
                    new AgentServiceEnabled { MappedServiceType = "VAVTranscript" }
                },
                Person = new Person
                {
                    EcePerson = false,
                    FirstName = DemoUtils.GenerateUniqueString("firstName", 1),
                    LastName = DemoUtils.GenerateUniqueString("lastName", 1),
                    LoginEnabled = true,
                    SsoEnabled = false,
                    UserName = DemoUtils.GenerateUniqueString("userName", 1),
                    Department = new ReferenceBean()
                },
                AgentDeskSettings = new ReferenceBean(),
                AgentTeam = new ReferenceBean(),
                Datacenter = new ReferenceBean(),
                DefaultSkillGroup = new ReferenceBean(),
                Department = new ReferenceBean(),
                Peripheral = new PeripheralRef(),
                PeripheralSet = new ReferenceBean()
            };

            // -- Create and get the agent
            var refUrl = await restClient.CreateAndGetAsync(newAgent);
            Agent? createdAgent = null;
            if (refUrl != null)
            {
                createdAgent = await restClient.GetAsync<Agent>(refUrl);
                Console.WriteLine($"Agent created: {createdAgent?.RefURL}");
                var services = createdAgent?.AgentServicesEnabled?.Select(s => s.MappedServiceType ?? "").ToList() ?? new List<string>();
                Console.WriteLine($"Agent Services enabled for the agent: {string.Join(", ", services)}");
            }

            // delete the agent
            if (createdAgent?.RefURL != null)
            {
                await restClient.DeleteAsync(createdAgent.RefURL);
                Console.WriteLine($"Deleted Agent: {createdAgent.RefURL}");
            }
        }
    }
}
